addons/mail/static/src/js/systray.js

```diff
@@ -68,6 +71,32 @@ var MessagingMenu = Widget.extend({
                     return channel.type !== 'static';
                 }
             });
+            var formatedFailures = [];
```

Comment:
What do you mean here?

Reply:
Don't store mailFailures as an attribute of systray. Simply get the mail failures on each update_channels_preview

So you can remove the line systray.js:49 `this.messageFailures = this.call('chat_manager', 'getMailFailures')`, and simply get the mail failures there.

Reply:
I can't really remove it line 49, i'm using it just after
var counter =  needactionCounter + unreadConversationCounter + this.messageFailures.length;

Reply:
(but I can switch it to a local variable)

Reply:
Okay for local variable in `update_counter`.

The idea of not having an attribute in systray for mail failures is to not rely on data that could potentially be outdated. Even if this is not the case now, this could become an issue in the future.